ls1intum/Artemis
# `Programming exercises`: Change Button Labels and add support for Local Testing and Feedback Mode

### Checklist
#### General


#### Server
-> no server changes made
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


https://github.com/ls1intum/Orion/issues/93/

### Description
Orion has received a patch that allows the usage of more IDEs next to IntelliJ. Therefore some of the Button labels weren't up to date anymore.
Furthermore Orion gets an update where students can view inline feedback directly in their IDE. For that a new Reload Button is introduced

### Steps for Testing
In Artemis:
1. Open a programming exercise! no extra button should be there

In [Orion](https://github.com/ls1intum/Orion)
1. Open a programming exercise
3. Test if the buttons are all appearing
2.1. Start assessment button
2.2. Open in your IDE button
2.3. Submit

No changes for Tutor buttons but they still should be visible

Without orion, all orion buttons should not be visible. (can be tested with a normal webbrowser)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage







### Screenshots



![grafik](https://github.com/ls1intum/Artemis/assets/25010075/21c2b40d-3bd4-449d-b2d1-2f7f0a42fc0c)

![grafik](https://github.com/ls1intum/Artemis/assets/25010075/ecdef516-9e48-44dd-9a40-efc36be75760)
